packages/apollo-language-server/src/languageProvider.ts

```diff
@@ -377,106 +372,4 @@ ${argumentNode.description}
 
     return null;
   }
-
-  async provideCodeLenses(
-    uri: DocumentUri,
-    _token: CancellationToken
-  ): Promise<CodeLens[]> {
-    const project = this.workspace.projectForFile(uri);
-    if (!project) return [];
-
-    await project.readyPromise;
-
-    const docsAndSets = project.documentsAt(uri);
-    if (!docsAndSets) return [];
-
-    let codeLenses: CodeLens[] = [];
-
-    for (const { doc, set } of docsAndSets) {
-      if (!doc.ast) continue;
-
-      for (const definition of doc.ast.definitions) {
-        if (definition.kind === Kind.OPERATION_DEFINITION) {
-          if (set.endpoint) {
-            const fragmentSpreads: Set<
-              graphql.FragmentDefinitionNode
-            > = new Set();
-            const searchForReferencedFragments = (node: graphql.ASTNode) => {
-              visit(node, {
-                FragmentSpread(node: FragmentSpreadNode) {
-                  const fragDefn = project.fragments[node.name.value];
-                  if (!fragDefn) return;
-
-                  if (!fragmentSpreads.has(fragDefn)) {
-                    fragmentSpreads.add(fragDefn);
-                    searchForReferencedFragments(fragDefn);
-                  }
-                }
-              });
-            };
-
-            searchForReferencedFragments(definition);
-
-            codeLenses.push({
-              range: rangeForASTNode(definition),
-              command: Command.create(
-                `Run ${definition.operation}`,
-                "apollographql.runQuery",
-                graphql.parse(
-                  [definition, ...fragmentSpreads]
-                    .map(n => graphql.print(n))
-                    .join("\n")
-                ),
-                definition.operation === "subscription"
-                  ? set.endpoint.subscriptions
-                  : set.endpoint.url,
-                set.endpoint.headers,
-                graphql.printSchema(set.schema!)
-              )
-            });
-          }
-        } else if (definition.kind === Kind.FRAGMENT_DEFINITION) {
-          const references = project.fragmentSpreadsForFragment(
```

Comment:
This part should probably be kept, it provides the `n references` codelens for fragments.

Reply:
Nice catch and thanks for taking a look!